Cov: Branch coverage (#3 #4 #5 #15)

fliljeda opened the pull request:
Added coverage for #3 #4 #5. In the command functions in check.py none of the branches are run in the tests but I do not have enough time to create tests for this as they are hard to call directly. However in utils/python.py there is one branch that is not run and a test can easily be modified to test this.

Fixes issue #3 #4 #5 #15

fliljeda committed:
Feat: Add coverage for the run function in commands/check.py

Very hard to make tests for this because they are very nested into the
code and is not easily called directly

Issue #4

fliljeda committed:
Feat: Add coverage for prepare_request callback function in parse.py

In commands/parse.py. Very hard to make tests for because it is nested
in code and hard to call directly

Issue #5

fliljeda committed:
Feat: Add coverage for get_func_args in utils/python.py

One branch is not entered and is easy to make test for. 1 new test
possible

Issue #3

dskeppstedt commented on scrapy/commands/check.py:
```diff
@@ -67,27 +71,57 @@ def run(self, args, opts):
-
+        
+        b1 = false
+        b2 = false
```
False
is the 
python 
way of
defining a
false boolean

dskeppstedt commented on scrapy/commands/check.py:
```diff
@@ -67,27 +71,57 @@ def run(self, args, opts):
+        b3 = false
+        b4 = false
         for spidername in args or spider_loader.list():
+            b1 = true
```
True
is the
python
way of defining
a true boolean

dskeppstedt commented on scrapy/commands/check.py:
```diff
@@ -67,27 +71,57 @@ def run(self, args, opts):
         if opts.list:
+            f.write("0005\n")
+            b1 = false
+            b2 = false
```
probably redo this to just write to the file instead as u did for the other functions

dskeppstedt reviewed (changes requested):
Add the reports and change some stuff plz

fliljeda committed:
Feat: Add coverage for exporters

One branch is not taken and could probably be added a test for

Issue #15

fliljeda committed:
Fix: Change true to True

Issue #4

fliljeda committed:
Fix: Change true to True and false to False

Issue #5

fliljeda committed:
Doc: Add report for prepare_request.callback

Is run when we run all tests in project. No test file is yet identified
but it misses one branch

Issue #5

fliljeda committed:
Cov: Changed coverage for else cases

Changed so that non-existant else cases exists and are counted as
branches

fliljeda committed:
Fix: add the correct amount of missed branches
